sympy/physics/mechanics/jointsmethod.py

```diff
@@ -0,0 +1,222 @@
+
+    Notes
+    =====
+
+    ``JointsMethod`` currently only works with systems that do not have any
+    configuration or motion constraints.
+
+    """
+
+    def __init__(self, newtonion, *joints):
+        if isinstance(newtonion, Body):
+            self.frame = newtonion.frame
+        else:
+            self.frame = newtonion
+
+        self._joints = joints
+        self._bodylist = self._generate_bodylist()
+        self._loadlist = self._generate_loadlist()
+        self._q = self._generate_q()
+        self._u = self._generate_u()
+        self._kdes = self._generate_kdes()
+
+        self._method = None
+
+    @property
+    def bodies(self):
+        return self._bodylist
+
+    @property
+    def forcelist(self):
+        return self._loadlist
+
+    @property
+    def q(self):
+        return self._q
+
+    @property
+    def u(self):
+        return self._u
+
+    @property
+    def kdes(self):
+        return self._kdes
+
+    @property
+    def forcing_full(self):
+        return self.method.forcing_full
+
+    @property
+    def mass_matrix_full(self):
+        return self.method.mass_matrix_full
+
+    @property
+    def mass_matrix(self):
+        return self.method.mass_matrix
+
+    @property
+    def forcing(self):
+        return self.method.forcing
+
+    @property
+    def method(self):
+        return self._method
+
+    def _generate_bodylist(self):
+        bodies = []
+        for joint in self._joints:
+            if joint.child not in bodies:
+                bodies.append(joint.child)
+            if joint.parent not in bodies:
+                bodies.append(joint.parent)
+        return bodies
+
+    def _generate_loadlist(self):
+        load_list = []
+        for body in self.bodies:
+            load_list.extend(body.loads)
+        return load_list
+
+    def _generate_q(self):
+        q_ind = []
+        for joint in self._joints:
+            q_ind.extend(joint.coordinates)
+        return q_ind
+
+    def _generate_u(self):
+        u_ind = []
+        for joint in self._joints:
+            u_ind.extend(joint.speeds)
+        return u_ind
+
+    def _generate_kdes(self):
+        kd_ind = []
+        for joint in self._joints:
+            kd_ind.extend(joint.kdes)
+        return kd_ind
+
+    def form_eoms(self, method=KanesMethod):
+        """Method to form system's equation of motions.
+
+        Parameters
+        ==========
+
+        method : Class
+            Class name of method.
+
+        Examples
+        ========
+
+        This is a simple example for a one degree of freedom translational
+        spring-mass-damper.
+
+        >>> from sympy.physics.mechanics import LagrangesMethod, dynamicsymbols, Body
+        >>> from sympy.physics.mechanics import PrismaticJoint, JointsMethod
+        >>> from sympy import symbols
+        >>> q = dynamicsymbols('q')
+        >>> qd = dynamicsymbols('q', 1)
+        >>> m, k, b = symbols('m k b')
+        >>> B = Body('B')
```

Comment:
make this consistent with the other example `wall = Body('W')` (or whatever you use above)